src/blob.rs

```diff
@@ -21,14 +64,33 @@ struct Header {
 
 impl<K> Blob<K>
 where
-    K: Default,
+    K: Default + Send,
 {
+    /// Creates new blob file or openes existing and truncates it
+    pub fn open_new<P: AsRef<Path> + Send + 'static>(path: P) -> BlobOpenFuture<K, P> {
```

Comment:
Since you defined `open` method, you shoud track the state of the `Blob` object. Currently it has two internal states: `Created` and `Opened`. If the `Blob` object in `Created` state then it should throw exception in `read` and `write` methods.